### PR TITLE
Aws user location and configure user role

### PR DESCRIPTION
**Pull request by nazarevitsch**

### CODE REVIEW

File: src/main/java/com/exadel/discount/platform/web/ImagesController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private AmazonService amazonService;
+
+    @PostMapping(value = "/upload_image",
+            consumes = {MediaType.MULTIPART_FORM_DATA_VALUE},
+            produces = {MediaType.APPLICATION_JSON_VALUE} )
+    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
+    public ResponseEntity<Message> upload(@RequestParam MultipartFile file){
+        try {
+            String link = amazonService.uploadFile(file);
+            log.info(link);
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/ImagesController.java:27]: debug.
And add some message describing what happened

File: src/main/java/com/exadel/discount/platform/web/UserDetailsController.java [hunk 2]
```diff
@@ -24,15 +25,15 @@
     private UserService userService;
 
     @GetMapping("/{user_id}")
-    @PreAuthorize("hasAnyRole('ADMINISTRATOR', 'USER')")
+    @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
     public ResponseEntity<?> getUserDetailsByUserId(@PathVariable("user_id")UUID userId, Principal principal) {
         if (userService.findUserByEmail(principal.getName()).getId().equals(userId)) {
             UserDetails userDetails = userDetailsService.findUserDetailsByUserId(userId);
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/UserDetailsController.java:31]: throw `NotFoundException` in the userDetailsService if there's no such user. Server will return 500 in that case but later we'll introduce ControllerAdvice to fix that

File: src/main/java/com/exadel/discount/platform/web/UserDetailsController.java [hunk 3]
```diff
@@ -32,7 +33,7 @@
             if (userDetails != null) {
                 return new ResponseEntity<>(userDetails, HttpStatus.OK);
             }
-            return new ResponseEntity<>(new Message("There isn't user details by this id!"), HttpStatus.NOT_FOUND);
+            throw new NotFoundException("User details wasn't found.");
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/UserDetailsController.java:36]: throw it in service, not here

File: src/main/java/com/exadel/discount/platform/web/ImagesController.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
     public ResponseEntity<Message> upload(@RequestParam MultipartFile file){
         try {
             String link = amazonService.uploadFile(file);
-            log.info(link);
+            log.debug(link);
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/ImagesController.java:27]: add a message describing what happened. e.g. "File was uploaded to amazon. Link: {}"

**Review by nkharitoniuk — APPROVED** [review]

**Pull request merged**